From 5538965d054838d3f3d19bb837b685f02525271b Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Wed, 12 Mar 2025 08:27:48 +0200
Subject: [PATCH] [SW-190031] boot_desc: use boot_desc_print_magic from cmn_lib

boot_desc_print_magic is the 1st functionality which
is moved to emb_cmn_lib, hence this patch removes
the local function and use the cmn function in emb_lib_cmn instead.

Change-Id: I31ddcf26e1a21066baf27766c6cf6b1e9bb3e03e
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/Makefile b/Makefile
index d68560f..414389d 100644
--- a/Makefile
+++ b/Makefile
@@ -48,6 +48,7 @@
 PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/common.c)
 PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/$(HL_TARGET).c)
 PPBOOT_SRCS_C += $(shell ls $(EMB_COMMON_MOD_DIR)/lib/common/lib_common.c)
+PPBOOT_SRCS_C += $(shell ls $(EMB_COMMON_MOD_DIR)/lib/common/boot_desc.c)
 PPBOOT_SRCS_C += $(shell ls $(EMB_COMMON_MOD_DIR)/lib/common/pci.c)
 PPBOOT_APP_S := $(shell ls app/ppboot/crt0_ppboot.S)
 PPBOOT_APP_S += $(shell ls $(HL_DEV_ARCH)/*.S | grep -v lds)
diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index ddd5abb..964786d 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -90,6 +90,7 @@
 	$(HL_DEV_ARCH_COMMON)/reset.o \
 	$(HL_DEV_ARCH)/$(HL_TARGET).o \
 	$(EMB_COMMON_MOD_DIR)/lib/common/lib_common.o \
+	$(EMB_COMMON_MOD_DIR)/lib/common/boot_desc.o \
 	$(EMB_COMMON_MOD_DIR)/lib/common/pci.o \
 	$(EMB_COMMON_MOD_DIR)/lib/preboot/fit_regions.o \
 	$(EMB_COMMON_MOD_DIR)/lib/preboot/fit_rsa.o \
diff --git a/lib/common/boot_desc.c b/lib/common/boot_desc.c
index b231ba8..d5a7e18 100644
--- a/lib/common/boot_desc.c
+++ b/lib/common/boot_desc.c
@@ -12,6 +12,7 @@
 #include <hl_fw_upd_if.h>		/* requires stdint.h */
 #include <hl_boot_desc.h>
 #include <common.h>
+#include <lib_common.h>
 #include <hl.h>
 
 #include "boot_desc.h"
@@ -25,26 +26,11 @@
 static u8 desc_buf[BOOT_DESC_MAX_IMGS * IMG_DESC_SZ];
 static bool desc_initialized;
 
-/* before removal and using emb_lib_common instead */
-static void __boot_desc_print_magic(char *name, struct hl_fw_boot_magic *magic)
-{
-	int i;
-
-	preboot_puts_raw(name);
-	preboot_puts_raw(" magic: ");
-	/* Strings do not obey endianness - print in reverse order */
-	for (i = ARRAY_SIZE(magic->name) - 1; i >= 0; i--)
-		preboot_putc_raw(magic->name[i]);
-
-	preboot_puts_hex_raw(" type: ", magic->type, false);
-	preboot_puts_hex(" value: ", magic->val, true);
-}
-
 __weak bool dev_spec_img_desc_is_valid(struct hl_fw_img_desc *img_desc) { return true; }
 
 static bool boot_desc_img_desc_is_valid(struct hl_fw_img_desc *img_desc)
 {
-	__boot_desc_print_magic("FW IMG DESC", &img_desc->magic);
+	boot_desc_print_magic("FW IMG DESC", &img_desc->magic);
 
 	if ((img_desc->magic.val & 0xFFFFFF00) != HL_FW_IMG_DESC_MAGIC)
 		return false;
@@ -128,7 +114,7 @@
 		if (boot_ptrs.boot_ptrs_sz > BOOT_PTRS_SZ)
 			preboot_puts("Warning: boot_ptrs size is bigger than expected\n");
 	}
-	__boot_desc_print_magic("FW boot PTRs", &boot_ptrs.magic);
+	boot_desc_print_magic("FW boot PTRs", &boot_ptrs.magic);
 
 	if (boot_ptrs.magic.val != HL_FW_BOOT_MAGIC)
 		return 0;
